app/models/concerns/preferences.rb

```diff
@@ -16,14 +16,7 @@ def preferences_for(preference, opts={})
 
     def scope_for(action, user, opts={})
       return all if user.is_admin?
-      case action.to_s
-      when "show", "index"
-        self.joins(:user, @preferences_for).where(:users => {private_profile: false},
-          @preferences_for.to_s.pluralize => {private: false})
-          .or(user.send("#{@preferences_for}_preferences"))
-      else
-        user.send("#{@preferences_for}_preferences")
-      end
+      where(user_id: user.id)
```

Comment:
No, it'll now just return the preferences for this user with FK filtering in the seriailzer via params. Note our doc's explicitly state that 'preferences are only visible to user they belong to'. So we aren't using it client side (to my knowledge) and I don't think we need this public project / private profile joining here at all. If we do need it reinstated we should figure out a better way then the joins here.